### PR TITLE
Integration test no longer modifies config paths

### LINKED ISSUE #189: integration test incorrectly modifies
this chunk modifies my `parameter_files/WebParameters_2dii.yml`, which makes the integration test fail, and causes more problems with my clone

https://github.com/2DegreesInvesting/PACTA_analysis/blob/ef5e55aa5fc3d2a3ff21dd6d3bda2c72b5fee861/integration-test.Rmd#L201-L230 

it changes my `parameter_files/WebParameters_2dii.yml` from:
```
default:
    paths:
        # Prefer "../PACTA_analysis/" -- not "./" nor "/path/to/PACTA_analysis"
        project_location_ext: ../PACTA_analysis/
        data_location_ext: ../pacta-data/2019Q4/
        template_location: ../create_interactive_report/
        stress_test_location: ../StressTestingModelDev/
    parameters:
        project_name: working_dir
        twodii_internal: FALSE
        new_data: FALSE
```

to
```
default:
    paths:
        # Prefer "../PACTA_analysis/" -- not "./" nor "/path/to/PACTA_analysis"
        project_location_ext: /Users/cj2dii/Documents/Git/PACTA_analysis"/
        data_location_ext: /Users/cj2dii/Documents/Git/pacta-data/2019Q4/
        template_location: /Users/cj2dii/Documents/Git/create_interactive_report/
        stress_test_location: /Users/cj2dii/Documents/Git/StressTestingModelDev/
    parameters:
        project_name: working_dir
        twodii_internal: FALSE
        new_data: FALSE
```

### PR DESCRIPTION
**Pull request by maurolepore**
Closes #189

### CODE REVIEW

File: integration-test.Rmd [hunk 1]
```diff
@@ -190,43 +190,29 @@ Ensure this other configuration file also exists:
 
 ```{r}
 config_2 <- here("parameter_files", "WebParameters_2dii.yml")
-config_2_copy <- tempfile()
-fs::file_copy(config_2, config_2_copy)
```
Comment (maurolepore) [inline integration-test.Rmd:194]: We no longer need to copy the configuration file, change it, and restore it at the end.

File: integration-test.Rmd [hunk 2]
```diff
@@ -190,43 +190,29 @@ Ensure this other configuration file also exists:
...
-    root_field_path("project_location_ext", pattern = "PACTA_analysis") %>%
-    root_field_path("data_location_ext", pattern = "pacta-data") %>%
-    root_field_path("template_location", pattern = "create_interactive_report") %>%
-    root_field_path("stress_test_location", pattern = "StressTestingModelDev")
-}
-
-root_field_path <- function(x, field, pattern) {
-  parent <- path_dir(here())
-  value <- path(parent, extract_from(x, pattern))
-  sub(glue("({field}:[ ]?).*"), glue("\\1{value}/"), x)
```
Comment (maurolepore) [inline integration-test.Rmd:221]: This was a lot of hacky code that I'm happy to remove :smile:

File: integration-test.Rmd [hunk 3]
```diff
@@ -190,43 +190,29 @@ Ensure this other configuration file also exists:
...
 
-extract_from <- function(x, pattern) {
-  line <- grep(pattern, x, value = TRUE)
-  sub(glue(".*({pattern}.*)"), "\\1", line)
+if (!uses_relative_paths(readLines(config_2))) {
+  abort("Pahts in the configuration file must use the form ../this_repo/")
 }
 
-make_config_portable(config_2)
+expect_true(uses_relative_paths(readLines(config_2)))
```
Comment (maurolepore) [inline integration-test.Rmd:215]: Here I'm checking that the paths use the form "../this_repo" -- this is explained in a comment on the configuration file, but a test is better than a comment :smile:

File: integration-test.Rmd [hunk 4]
```diff
@@ -321,12 +307,3 @@ In read_file(paste0(file_location, "/fund_data.fst")) :
...
 ```
-
-## Cleanup
-
-Restore configuration file.
-
-```{r}
-fs::file_copy(config_2_copy, config_2, overwrite = TRUE)
-```
-
```
Comment (maurolepore) [inline integration-test.Rmd:332]: No need for cleanup.

**Review by maurolepore — COMMENTED** [review]
This PR improves the code base by reducing the code, 
and by adding a test for a requirement that before was
only explained as a comment.

File: integration-test.Rmd [hunk 5]
```diff
@@ -190,43 +190,29 @@ Ensure this other configuration file also exists:
...
+  
+  all(purrr::map_lgl(patterns, ~any(grepl(.x, lines))))
 }
 
-root_field_path <- function(x, field, pattern) {
-  parent <- path_dir(here())
-  value <- path(parent, extract_from(x, pattern))
-  sub(glue("({field}:[ ]?).*"), glue("\\1{value}/"), x)
+if (!uses_relative_paths(readLines(config_2))) {
+  abort("Pahts in the configuration file must use the form ../this_repo/")
```
Comment (cjyetman) [inline integration-test.Rmd:212]: spelling "Paths"

**Comment by cjyetman** [comment]
honestly, I haven't fully wrapped my head around this, but on the surface it seems to make sense

pretty sure line205:`writeLines(lines, config)` was the thorn in my side, so happy that's gone :-)

**Pull request merged**